src/buttons/RolesEditionButton/RolesEditionButton.js

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Button, IconButton, Tooltip } from '@material-ui/core';
+import ShareIcon from '@material-ui/icons/Share';
+import RolesEditionDialog from './components/RolesEditionDialog/RolesEditionDialog';
+
+export const RolesEditionButton = ({
+  labels,
+  isIconButton,
+  users,
+  scenarioRolesPermissionsMapping,
```

Comment:
Would it be possible to abstract the "scenario" concept here and replace it by "resource" ? Or do we have some behavior in the component that is specific to scenario management ?

Reply:
Good idea ! The component is indeed working with an array of roles, there is nothing specific to scenario :slightly_smiling_face: